RecoTracker/PixelLowPtUtilities/src/StripSubClusterShapeTrajectoryFilter.cc

```diff
@@ -265,74 +265,84 @@ bool StripSubClusterShapeFilterBase::testLastHit(const TrackingRecHit *hit,
-    while (hitStripsTrim > 1 && (*begin < std::max<uint8_t>(trimCut, trimMaxFracNeigh_ * (*(begin + 1))))) {
-      hitStripsTrim--;
-      ++begin;
-    }
-    while (hitStripsTrim > 1 && (*last < std::max<uint8_t>(trimCut, trimMaxFracNeigh_ * (*(last - 1))))) {
-      hitStripsTrim--;
-      --last;
-    }
+      // trimming
+      INC_COUNTER(test_)
+      unsigned int hitStripsTrim = ampls.size();
+      int sum = std::accumulate(ampls.begin(), ampls.end(), 0);
+      uint8_t trimCut = std::min<uint8_t>(trimMaxADC_, std::floor(trimMaxFracTotal_ * sum));
+      auto begin = ampls.begin();
+      auto last = ampls.end() - 1;
+      while (hitStripsTrim > 1 && (*begin < std::max<uint8_t>(trimCut, trimMaxFracNeigh_ * (*(begin + 1))))) {
+        hitStripsTrim--;
+        ++begin;
+      }
+      while (hitStripsTrim > 1 && (*last < std::max<uint8_t>(trimCut, trimMaxFracNeigh_ * (*(last - 1))))) {
+        hitStripsTrim--;
+        --last;
+      }
 
-    if (hitStripsTrim < std::floor(std::abs(hitPredPos) - maxTrimmedSizeDiffNeg_)) {
-      INC_COUNTER(failTooNarrow_)
-      return false;
-    } else if (hitStripsTrim <= std::ceil(std::abs(hitPredPos) + maxTrimmedSizeDiffPos_)) {
-      INC_COUNTER(passTrim_)
-      return true;
-    }
+      if (hitStripsTrim < std::floor(std::abs(hitPredPos) - maxTrimmedSizeDiffNeg_)) {
+        INC_COUNTER(failTooNarrow_)
+        return false;
+      } else if (hitStripsTrim <= std::ceil(std::abs(hitPredPos) + maxTrimmedSizeDiffPos_)) {
+        INC_COUNTER(passTrim_)
+        return true;
+      }
 
-    const StripGeomDetUnit *stripDetUnit = dynamic_cast<const StripGeomDetUnit *>(det);
-    if (det == nullptr) {
-      edm::LogError("Strip not a StripGeomDetUnit?") << " on " << detId.rawId() << "\n";
-      return true;
-    }
+      const StripGeomDetUnit *stripDetUnit = dynamic_cast<const StripGeomDetUnit *>(det);
+      if (det == nullptr) {
+        edm::LogError("Strip not a StripGeomDetUnit?") << " on " << detId.rawId() << "\n";
+        return true;
+      }
 
-    float MeVperADCStrip = 9.5665E-4;  // conversion constant from ADC counts to MeV for the SiStrip detector
-    float mip =
-        3.9 / (MeVperADCStrip / stripDetUnit->surface().bounds().thickness());  // 3.9 MeV/cm = ionization in silicon
-    float mipnorm = mip / std::abs(ldir.z());
-    ::SlidingPeakFinder pf(std::max<int>(2, std::ceil(std::abs(hitPredPos) + subclusterWindow_)));
-    ::PeakFinderTest test(mipnorm,
-                          detId(),
-                          cluster.firstStrip(),
-                          &*theNoise,
-                          seedCutMIPs_,
-                          seedCutSN_,
-                          subclusterCutMIPs_,
-                          subclusterCutSN_);
-    if (pf.apply(cluster.amplitudes(), test)) {
-      INC_COUNTER(passSC_)
-      return true;
+      float MeVperADCStrip = 9.5665E-4;  // conversion constant from ADC counts to MeV for the SiStrip detector
```

Comment:
this is the same number you use at

https://github.com/cms-sw/cmssw/blob/106a6049207c16e3e93003333a86305c8c739cdf/RecoLocalTracker/SiStripClusterizer/plugins/SiStripClusters2ApproxClusters.cc#L99

perhaps it should be actually centralized.

Reply:
Comment not addressed ?